internal/provider/kubernetes/gatewayclass.go

```diff
@@ -103,7 +103,11 @@ func (r *gatewayClassReconciler) Reconcile(ctx context.Context, request reconcil
 
 	for i := range gatewayClasses.Items {
 		if gatewayClasses.Items[i].Spec.ControllerName == r.controller {
-			cc.addMatch(&gatewayClasses.Items[i])
+			if !gatewayClasses.Items[i].DeletionTimestamp.IsZero() {
+				cc.removeMatch(&gatewayClasses.Items[i])
```

Comment:
shouldnt `removeMatch` be added https://github.com/envoyproxy/gateway/blob/96e7f5495cb8d2006178b981c70febd32ae01efa/internal/provider/kubernetes/gatewayclass.go#L124 as well as move it before https://github.com/envoyproxy/gateway/blob/96e7f5495cb8d2006178b981c70febd32ae01efa/internal/provider/kubernetes/gatewayclass.go#L115 ?

Reply:
Resolved by https://github.com/envoyproxy/gateway/pull/421/commits/82fb860dc966dd0e5ba05781d49a0dd69ce47fc2